### ISSUE TITLE
"panic: assignment to entry in nil map" error

### ISSUE THREAD

**Issue by qingling128** [census-ecosystem/opencensus-go-exporter-stackdriver#265]
Please answer these questions before submitting a bug report.

### Error

```
level=info ts=2020-03-02T03:17:42.590Z caller=main.go:293 msg="Starting Stackdriver Prometheus sidecar" version="(version=0.7.3, branch=master, revision=2548e8a7b383adc9217afe8570247ea84e3ef2c0)"
level=info ts=2020-03-02T03:17:42.591Z caller=main.go:294 build_context="(go=go1.12, user=kbuilder@kokoro-gcp-ubuntu-prod-40088225, date=20200206-15:24:17)"
level=info ts=2020-03-02T03:17:42.591Z caller=main.go:295 host_details="(Linux 4.14.138+ #1 SMP Tue Sep 3 02:58:08 PDT 2019 x86_64 apigee-metrics-v111-fd6fj (none))"
level=info ts=2020-03-02T03:17:42.592Z caller=main.go:296 fd_limits="(soft=1048576, hard=1048576)"
panic: assignment to entry in nil map

goroutine 1 [running]:
contrib.go.opencensus.io/exporter/stackdriver.NewExporter(0x7ffca84e2165, 0x16, 0xc0001fc13d, 0xd, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
        /usr/local/go/src/github.com/Stackdriver/stackdriver-prometheus-sidecar/vendor/contrib.go.opencensus.io/exporter/stackdriver/stackdriver.go:343 +0x437
main.main()
        /usr/local/go/src/github.com/Stackdriver/stackdriver-prometheus-sidecar/cmd/stackdriver-prometheus-sidecar/main.go:332 +0x5ca9
```

### What version of the Exporter are you using?
https://github.com/Stackdriver/stackdriver-prometheus-sidecar/blob/2548e8a7b383adc9217afe8570247ea84e3ef2c0/vendor/contrib.go.opencensus.io/exporter/stackdriver/stackdriver.go#L343

### What version of Go are you using?
1.12

### What did you expect to see?
An error that is more informative or no error.

### What did you see instead?
An "panic: assignment to entry in nil map" error

### Additional context
We are thinking about sending a PR for a fix. But it's not clear what the expected behavior is. 

Are these env variables supposed to always be present? https://github.com/census-instrumentation/opencensus-go/blob/1901b56b9515b0c34f5d25a5bce982dfc543d64b/resource/resource.go#L30

Are the resource type and labels supposed to be always initiated in the resource detector (https://github.com/Stackdriver/stackdriver-prometheus-sidecar/blob/2548e8a7b383adc9217afe8570247ea84e3ef2c0/vendor/contrib.go.opencensus.io/exporter/stackdriver/stackdriver.go#L337)?

Is something like this below a reasonable fix for https://github.com/census-ecosystem/opencensus-go-exporter-stackdriver/blob/1905f4643248273f1ed512fdf1b6473d2e4445b4/stackdriver.go#L347?

```
		// Populate internal resource labels for defaulting project_id, location, and
		// generic resource labels of applicable monitored resources.			// generic resource labels of applicable monitored resources.
+		if res.Labels == nil {
+			res.Labels = make(map[string]string)
+		}
		res.Labels[stackdriverProjectID] = o.ProjectID			res.Labels[stackdriverProjectID] = o.ProjectID
		res.Labels[resourcekeys.CloudKeyZone] = o.Location			res.Labels[resourcekeys.CloudKeyZone] = o.Location
		res.Labels[stackdriverGenericTaskNamespace] = "default"			res.Labels[stackdriverGenericTaskNamespace] = "default"
```

**Comment by qingling128** [comment]
Should we only set those labels (https://github.com/census-ecosystem/opencensus-go-exporter-stackdriver/blob/1905f4643248273f1ed512fdf1b6473d2e4445b4/stackdriver.go#L349) if the monitored resource turned out to be generic_node or generic_task?

**Comment by StevenYCChou** [comment]
Given the comment on
https://github.com/census-instrumentation/opencensus-go/blob/1901b56b9515b0c34f5d25a5bce982dfc543d64b/resource/resource.go#L137-L138

I think returning nil seems correct.

**Comment by qingling128** [comment]
Sent https://github.com/census-ecosystem/opencensus-go-exporter-stackdriver/pull/264.

**Comment by nilebox** [comment]
Fixed by #264, closing.
